code/code/sys/configuration.cc

```diff
@@ -183,3 +186,30 @@ bool Config::doConfiguration(int argc, char* argv[]) {
   }
   return true;
 }
+
+std::string readFileContents(const std::string& path) {
+  std::ifstream file(path);
+
+  if (!file.is_open()) {
+    vlogf(LOG_FILE,
+      format("Failed to open %s.") % (std::filesystem::current_path() / path));
+    return {};
+  }
+
+  return {(std::istreambuf_iterator<char>(file)), {}};
+}
+
+std::string readVersionFromFile() {
+  const std::filesystem::path versionPath =
+    std::filesystem::current_path() / Config::DataDir().c_str() / "version.txt";
```

Comment:
If desired, you could avoid setting this path by moving the reading of version to happen after `if (chdir(Config::DataDir().c_str())` in `main()`.

Reply:
I was trying to keep the line that initializes MUD_NAME_VERS by calling this function as high up in `main()` as possible, so that it's the first output one sees when starting the boot process (since that's how you had it set up previously).